File: stack&queue/programmers_42583/solution.py
def solution(bridge_length, weight, truck_weights):
    time = 0
    bridge = [0] * bridge_length
    
    while bridge:
        bridge.pop(0)
        time += 1
        
        if truck_weights:
            bridge_weight = sum(bridge)
            
            if(bridge_weight + truck_weights[0] <= weight):
                bridge.append(truck_weights.pop(0))
            else:
                bridge.append(0)
        
    return time

File: stack&queue/programmers_42583/solution.js
function solution(bridge_length, weight, truck_weights) {
    let time = 0;
    let bridge = Array(bridge_length).fill(0);
    
    while(bridge.length) {
        bridge.shift();
        time ++;
        
        if(truck_weights.length){
            let bridge_weight = bridge.reduce((a, b) => a + b);
            
            if(bridge_weight + truck_weights[0] <= weight){
                bridge.push(truck_weights.shift());
            } else {
                bridge.push(0);
            }
        }
    }
    
    return time;    
}